### DISCUSSION TITLE
Simple drawing issues

### DISCUSSION THREAD

**Discussion by Pickle** [open-watcom/open-watcom-v2#971] [Q&A] [answered]
ive been trying out the compiler on dos and the drawing routines behavior doesnt make any sense.
I really just wanted to draw eclipses across the screen. Usually After the first 2 shapes things just go nuts.

`wcl -zq -we -l=dos main.c`

```

#include <stdio.h>
#include <stdint.h>
#include <graph.h>
#include <conio.h>

#define SCREEN_W 640
#define SCREEN_H 480
#define BLOCKS_W 10
#define BLOCKS_H 10

void pause( void )
{
  if (getch() == 0) {
     getch();
  }
}

int main( void )
{
   int result;
   uint16_t x = 0;
   uint16_t y = 0;
   uint16_t block_w = SCREEN_W / BLOCKS_W;
   uint16_t block_h = SCREEN_H / BLOCKS_H;
   uint8_t color = 1;

   result = _setvideomode( _VRES16COLOR );
   if (result == 0)
   {
      printf( "Error could not open video. code %d\n", result );
      return 1;
   }

   for (y=0; y<SCREEN_H; y+=block_h)
   {
      for (x=0; x<SCREEN_W; x+=block_w)
      {
         //printf( "x %d y %d w %d h %d\n", x, y, block_w, block_h );

         _setcolor( color );
         _ellipse( _GFILLINTERIOR, x, y, block_w, block_h );

         color++;
         if (color > 15)
            color = 1;

         //pause();
      }
   }

   pause();

   _setvideomode( _DEFAULTMODE );

   return 0;
}

```

Ive even tried simple like drawing 4 rectangles at 
0,0,50,50
100,100,50,50
200,200,50,50
300,300,50,50
First 2 are ok but the third and fourth just explode in size. I dont get it, any help appreciated cause i thought i knew how to draw rects ;-)

**Comment by Pickle** [comment, marked as answer]
figured out finally. The arguments for the function is x1,y1,x2,y2.
`_ellipse( _GFILLINTERIOR, x, y, x+block_w, y+block_h );`

It doesnt help the header file only has has the types in the declarations so it a guessing game what things are.
I figured out by finally finding where the function lives in the right C file.
Tried finding it in the wiki info but i havnt seen it. Doesnt seem very ideal.

**Reply by jmalak** [reply]
I am not sure where you try to get info, here is standard documentation where **[_ellipse](https://open-watcom.github.io/open-watcom-v2-wikidocs/clib.html#_ellipse)** is described.
Anyway **[general description of Graphics system](https://open-watcom.github.io/open-watcom-v2-wikidocs/clib.html#Graphics_Library)** is there too.

**Reply by Pickle** [reply]
I dont know how i missed that page but thanks for pointing that out thats what im looking for.

**Reply by jmalak** [reply]
You are welcome
